src/modules/cpu/kernel/transpose_generic.hpp

```diff
@@ -0,0 +1,482 @@
+                                    dstPtr3++;
+                                }
+                                srcPtr2 += 1;
+                                dstPtr2 += dstGenericDescPtr->strides[3];
+                            }
+                            srcPtr1 += srcGenericDescPtr->strides[3];
+                            dstPtr1 += dstGenericDescPtr->strides[2];
+                        }
+                        srcPtr0 +=  srcGenericDescPtr->strides[2];
+                        dstPtr0 += dstGenericDescPtr->strides[1];
+                    }
+                }
+            }
+            else
+            {
+                // compute output shape
+                for(Rpp32u i = 0; i < nDim; i++)
+                    dstShape[i] = length[perm[i]];
+
+                // compute output strides
+                compute_strides(dstStrides, dstShape, nDim);
+
+                // compute input strides and update as per the permute order
+                Rpp32u tempStrides[RPPT_MAX_DIMS];
+                compute_strides(tempStrides, roi, nDim);
+                for(int i = 0; i < nDim; i++)
+                    srcStrides[i] = tempStrides[perm[i]];
+
+                // perform transpose as per the permute order
+                transpose(dstPtrTemp, dstStrides, srcPtrTemp, srcStrides, dstShape, nDim);
+            }
+        }
+    }
+    free(srcStridesTensor);
+    free(dstStridesTensor);
+    free(dstShapeTensor);
+
+    return RPP_SUCCESS;
+}
+
+template<typename T>
+RppStatus transpose_generic_host_tensor(T *srcPtr,
+                                        RpptGenericDescPtr srcGenericDescPtr,
+                                        T *dstPtr,
+                                        RpptGenericDescPtr dstGenericDescPtr,
+                                        Rpp32u *permTensor,
+                                        Rpp32u *roiTensor,
+                                        RppLayoutParams layoutParams,
+                                        rpp::Handle& handle)
+{
+    Rpp32u numThreads = handle.GetNumThreads();
+    Rpp32u nDim = dstGenericDescPtr->numDims;
+    Rpp32u batchSize = dstGenericDescPtr->dims[0];
+
+    // allocate buffer for input strides, output strides, output shapes
+    Rpp32u *srcStridesTensor = (Rpp32u *)calloc(nDim * batchSize, sizeof(int));
+    Rpp32u *dstStridesTensor = (Rpp32u *)calloc(nDim * batchSize, sizeof(int));
+    Rpp32u *dstShapeTensor = (Rpp32u *)calloc(nDim * batchSize, sizeof(int));
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < batchSize; batchCount++)
+    {
+        T *srcPtrTemp, *dstPtrTemp;
+        srcPtrTemp = srcPtr + batchCount * srcGenericDescPtr->strides[0];
+        dstPtrTemp = dstPtr + batchCount * dstGenericDescPtr->strides[0];
+
+        Rpp32u *dstShape = dstShapeTensor + batchCount * nDim;
+        Rpp32u *roi = roiTensor + batchCount * nDim * 2;
```

Comment:
Can add some code-block level comments to clarify what logically separated code blocks do

Reply:
Add comments for getting start and begin value from ROI tensor